lib/pengine/unpack.c

```diff
@@ -89,14 +89,20 @@ pe_fence_node(pe_working_set_t * data_set, node_t * node, const char *reason)
         set_bit(node->details->remote_rsc->flags, pe_rsc_failed);
 
     } else if (is_baremetal_remote_node(node)) {
-        if(pe_can_fence(data_set, node)) {
-            crm_warn("Node %s will be fenced %s", node->details->uname, reason);
+        resource_t *rsc = node->details->remote_rsc;
+
+        if ((rsc != NULL) && (is_set(rsc->flags, pe_rsc_managed))) {
```

Comment:
A remote node can have no resource if its resource was removed from the configuration (orphaned). It's unlikely that the node would be fenced in such a case, but we should allow the possibility, so this should be (rsc == NULL) || ...

If that looks odd, just flip the blocks so we use if (rsc && is...managed) "Not fencing" else ...